src/extensions/posthog.ts

```diff
@@ -9,26 +12,67 @@ export interface DummyPostHog {
 export function createPosthog(server: PluginsServer, pluginConfig: PluginConfig): DummyPostHog {
     const distinctId = pluginConfig.plugin?.name || `plugin-id-${pluginConfig.plugin_id}`
 
-    function sendEvent(event: string, properties: Record<string, any> = {}) {
-        const client = new Client(server.redis, server.PLUGINS_CELERY_QUEUE)
-
-        const data = {
-            distinct_id: distinctId,
-            event,
-            timestamp: new Date(),
-            properties: {
-                $lib: 'posthog-plugin-server',
-                $lib_version: version,
-                ...properties,
-            },
-        }
-
-        client.sendTask(
-            'posthog.tasks.process_event.process_event_with_plugins',
-            [distinctId, null, null, data, pluginConfig.team_id, new Date(), new Date()],
-            {}
-        )
-    }
+    const producer = server.EE_ENABLED ? server.kafka!.producer() : null
+    producer?.connect()
+
+    const sendEvent = server.EE_ENABLED
```

Comment:
nit, but with `? :` blocks that huge, it's worth using an if loop, especially due to this `producer` stuff in front